Directory Dialog

Davischw26 opened the pull request:
I have implemented a new feature in the ImageRater application that allows the user to select a directory when the application starts. To accomplish this, I integrated Windows Forms into the application.

Davischw26 committed:
Directory Dialog

g4m3r0 commented on ImageRater/MainWindow.xaml.cs:
```diff
@@ -1,18 +1,28 @@
-        this.DataContext = this.ImageRater;
+        using (var dialog = new FolderBrowserDialog())
+
+            while (!Directory.Exists(dialog.SelectedPath))
```
Do you think that the while loop is necessary?
I think the dialogs already checks if the folder exists on its own.

g4m3r0 commented on ImageRater/MainWindow.xaml.cs:
```diff
@@ -1,18 +1,28 @@
     {
-        this.ImageRater = new("C:\\workingDirectory\\ForschungsPraktikum\\D600\\screenshots\\screenshots\\screen_all_in_one_folder");
-        this.DataContext = this.ImageRater;
+        using (var dialog = new FolderBrowserDialog())
```
Can you give the dialog a descriptive title?
Maybe something like "Please select your image input folder".

g4m3r0 commented:
Hey @Davischw26,
I wanted to thank you for your commit. Although I must admit that I'm not a big fan of using Windows Forms with WPF, you are correct that this is currently the recommended solution according to this GitHub issue (https://github.com/dotnet/wpf/issues/438).

I have reviewed your submitted code and left two comments for you to consider. Thank you again for your contribution!
#3

Davischw26 committed:
Improvement File Dialog

Davischw26 commented:
I agree that the while loop wasn't necessary. Instead, I modified the code to handle the case where the user exits the dialog without selecting a directory by closing the program. I also added the descriptive title to the dialog and an event handler that brings the window to the front after the user has selected a directory.

g4m3r0 commented on ImageRater/MainWindow.xaml.cs:
```diff
@@ -14,16 +13,30 @@ public partial class MainWindow : Window
     }
+
+    // This event handler is called after the MainWindow's content has been rendered. 
+    // It sets the Topmost property of the MainWindow to true and then false, 
```
It is better to use the XML-based comments for documentation e.g.:

```
    /// <summary>
    ///  Determines if a string is fancy or not.
    /// </summary>
    /// <param name="MyParameter">This is a fancy parameter.</param>
   /// <returns>Returns True if the parameter is fance, otherwise False.</returns>
```

g4m3r0 commented on ImageRater/MainWindow.xaml.cs:
```diff
@@ -14,16 +13,30 @@ public partial class MainWindow : Window
-
-            while (!Directory.Exists(dialog.SelectedPath))
+        {
+            dialog.Description = "Please select the folder that contains your input images.";
```
Looks good tho :).
You can also directly declare the parameters when creating the instance:
```
        using (var dialog = new FolderBrowserDialog()
        {
            Description = "Please select the folder that contains your input images.",
            UseDescriptionForTitle = true,
        })
```
This is just my personal taste, it makes the declaration more readable imho.

g4m3r0 commented on ImageRater/MainWindow.xaml.cs:
```diff
@@ -14,16 +13,30 @@ public partial class MainWindow : Window
     public MainWindow()
     {
         using (var dialog = new FolderBrowserDialog())
```
I think it would be good if the folder dialog would get it's own function inside the window code behind (this file) which returns the selected folder path, to make the code better readable.

g4m3r0 commented:
Hey @Davischw26,
again thank you for your commit :).

I have reviewed your submitted code and added some comments for potential improvements.
Tell me what you think about that.
